trlx/models/modeling_nemo_ilql.py

```diff
@@ -783,4 +756,30 @@ def generate(
                 "compute_logprob": False,
             }
 
-        return super().generate(inputs, length_params, sampling_params)
+        sp_was_enabled = self.cfg.get("sequence_parallel", False)
+        if sp_was_enabled:
+            self.sequence_parallel_(False)
+
+        activations_checkpointing_was_enabled = self.cfg.get("activations_checkpoint_granularity", None) is not None
+
+        if activations_checkpointing_was_enabled:
+            self.activation_checkpointing_(False)
+
+        gen = super().generate(inputs, length_params, sampling_params)
+
+        if activations_checkpointing_was_enabled:
+            self.activation_checkpointing_(True)
+
+        if sp_was_enabled:
```

Comment:
Why do you set it twice if sp was enabled? first to false then to true haha.

Reply:
disable if enabled and then re-enable if enabled